ibc-apps/ics721-nft-transfer/src/handler/send_transfer.rs

```diff
@@ -81,14 +82,20 @@ where
                 &msg.chan_id_on_a,
                 class_id,
                 token_id,
-                &packet_data.memo.clone().unwrap_or_default(),
+                &packet_data
+                    .memo
+                    .clone()
+                    .unwrap_or(Memo::from("".to_string())),
             )?;
         } else {
             transfer_ctx.burn_nft_validate(
                 &sender,
                 class_id,
                 token_id,
-                &packet_data.memo.clone().unwrap_or_default(),
+                &packet_data
+                    .memo
+                    .clone()
+                    .unwrap_or(Memo::from("".to_string())),
```

Comment:
Since this refactor became longer, you may change the `FromStr` implementation for Memo to `From<&str>` and use `Memo::from("")` here.

Reply:
@Farhad-Shabani afaik `FromStr` and `TryFrom<&str>` are similar. Moreover, `FromStr` is used for parsing, whereas `From` is for conversion. For example;

```rs
u64::from_str("23"); // Ok(23_u64)
String::from("hello world"); // "hello world".to_owned()
```

Reply:
Yeah, on the same page with you. I meant using a more idiomatic approach, but, it's no biggie.